File: android/app/src/main/kotlin/com/tripay/data/SyncEngine.kt
package com.tripay.data

import com.tripay.models.Transaction
import java.util.Date

class SyncEngine {
    fun resolveConflict(local: Transaction, remote: Transaction): Transaction {
        return if (local.date.after(remote.date)) local else remote
    }

    suspend fun performSync() {
        // Background sync logic using WorkManager
        println("Syncing data in background...")
    }
}

File: ios/TriPay/Data/SyncEngine.swift
import Foundation

class SyncEngine {
    static let shared = SyncEngine()
    
    func resolveConflict(local: Transaction, remote: Transaction) -> Transaction {
        // Simple timestamp-based resolution
        if local.date > remote.date {
            return local
        } else {
            return remote
        }
    }
    
    func performSync() async throws {
        // Logic for background synchronization
        print("Performing background sync...")
    }
}
